### PR TITLE
Delete push tokens on logout, send push tokens on login

### PR DESCRIPTION
**Pull request by chromakode**
@mmaxim Would you please review the CORE changes (added HTTP DELETE method)?

### CODE REVIEW

File: go/service/apiserver.go [hunk 1]
```diff
@@ -118,6 +122,16 @@ func (a *APIServerHandler) doPostJSON(rawarg keybase1.PostJSONArg) (res keybase1
 	return a.convertRes(ires), nil
 }
 
+func (a *APIServerHandler) doDelete(arg keybase1.DeleteArg) (res keybase1.APIRes, err error) {
+	a.G().Trace("APIServerHandler::Post", func() error { return err })()
```
Comment (mmaxim) [inline go/service/apiserver.go]: Changed this to `Delete`

Comment (chromakode) [inline go/service/apiserver.go]: Oops! Good spot.

**Comment by patrickxb** [comment]
this got merged in with a compilation error

**Comment by maxtaco** [comment]
@chromakode this change doesn't compile, it breaks the test mocks in libkb.

**Pull request merged**